### ISSUE TITLE
Feature request: Vertical tabs

### ISSUE THREAD

**Issue by bstst** [zed-industries/zed#4331]
### Check for existing issues

- [X] Completed

### Describe the feature

I would love to have the ability to switch the tab layout to vertical.

just like JetBrains do. this feature request is somewhat related to https://github.com/zed-industries/community/issues/459, yet it's not a request for a separate pane, only the ability to switch tab layout.

### If applicable, add mockups / screenshots to help present your vision of the feature

![image](https://github.com/zed-industries/community/assets/412859/96ab233c-122b-4d1f-b81f-c2c5f5b9438d)

**Comment by syntastical** [comment]
Yeah, probably seems trivial to most, but vertical tabs has become a crucial part of my IDE needs.

**Comment by omid** [comment]
I'm just playing around with the source code to get more familiar with it.
It's what I have so far, but I'm not sure if the zed devs will accept the changes or not, or how generally the process (I've read the CONTRIBUTING file) works? It's not a tiny change, it's also not a big one!

So far, it's like the screenshot below, but it's not resizable, and it's just _top_ or _right_ (and _none_, to remove one config? and I will try to add left and bottom also).
It's not a `pane`, it's exactly the same `tab_bar` we have on top.

Ref: https://github.com/zed-industries/zed/discussions/6667#discussioncomment-10025781 (and a bit https://github.com/zed-industries/zed/issues/5240)

![image](https://github.com/user-attachments/assets/e9ee2082-a85b-447e-951f-be63a710bfbc)

PS: the code is not close to ready to be a PR.

**Comment by m-czyz** [comment]
hey @omid could you somehow share your WIP changes? I would like to contribute to the development of this feature too, but I don't want to start from the scratch

**Comment by omid** [comment]
@m-czyz sure. But be aware that it's just a PoC.
For example, I completely commented out the top bar, instead of handling it by conditions.
I made a PR [here](https://github.com/omid/zed/pull/1) for [this](https://github.com/omid/zed/tree/feat/vertical_bar) branch.

PS: We can remove `tab_bar.show` config also. And add a `None` to `TabBarPosition` enum.
